wled00/wled_math.cpp

```diff
@@ -88,8 +88,8 @@ uint8_t cos8_t(uint8_t theta) {
 float sin_approx(float theta)
 {
   theta = modd(theta, TWO_PI); // modulo: bring to -2pi to 2pi range
-  if(theta < 0) theta += M_TWOPI; // 0-2pi range
-  uint16_t scaled_theta = (uint16_t)(theta * (0xFFFF / M_TWOPI));
+  if(theta < 0) theta += TWO_PI; // 0-2pi range
```

Comment:
Why? Why not change `M_PI` below?

Reply:
good question. I think this was auto-complete suggestion from co-pilot. 
I will make it consistant to M_PI variants.
Also: the conversion of the angle using float is really not needed, I added that to fix a bug. but the actual fix is: cast a negative float to a `int` and NEVER to a `unsigned` type, as that behaviour is undefined and depends on the CPU/Compiler, hence the issues on the C3 which does `(uint)(float) = 0`